hs_core/models.py

```diff
@@ -1349,14 +1363,29 @@ class Meta:
         unique_together = ("content_type", "object_id")
 
 def get_path(instance, filename):
-    return os.path.join(instance.content_object.short_id, 'data', 'contents', filename)
+    if instance.content_object.resource_federation_path:
+        return os.path.join(instance.content_object.resource_federation_path, instance.content_object.short_id, 'data',
+                            'contents', filename)
+    else:
+        return os.path.join(instance.content_object.short_id, 'data', 'contents', filename)
 
 class ResourceFile(models.Model):
```

Comment:
@hyi Let me try to explain what I meant by 'abstract away the location of the content file':

Here is an use case how the location of the resource file possibly can be abstracted away:

The following block of code has been duplicated in all resource specific ```receivers.py``` where there is a signal handler function for resource file pre-delete action. As we can see the code logic is very dependent on the location of the file in order to delete it.

    for f in ResourceFile.objects.filter(object_id=res.id):
       if f.resource_file and f.resource_file.name != res_fname: 
          f.resource_file.delete() 
          f.delete() 
      elif f.fed_resource_file and f.fed_resource_file != res_fname: 
          f.fed_resource_file.delete() 
          f.delete()
      elif f.fed_resource_file_name_or_path and f.fed_resource_file_name_or_path != res_fname: 
          file_name = os.path.join(res.resource_federation_path, res.short_id, f.fed_resource_file_name_or_path)
          utils.delete_fed_zone_file(file_name) 
          f.delete()

Here the various resource types has to be aware of the location of the file in order to delete it. Instead it would be nice if any resource type can do the file deletion using a code pattern like the following:

    for f in ResourceFile.objects.filter(object_id=res.id): 
       utils.delete_resource_file(f)
    
OR

      for f in ResourceFile.objects.filter(object_id=res.id): 
          ResourceFile.delete(f)

Similar to the file delete operation described above there are other file operations (like copying the file to a temp directory - needed for file I/O, getting the name of the file, getting the extension of the file) code logic which is again dependent on the location of the file (local zone or federated zone) and some of the resource types have duplicated that logic.